### PR TITLE
continuing discussing on macros -> UDFs (fixing and testing with run-operations)

### PR DESCRIPTION
**Pull request by andrewhong5297**
Brief comments on the purpose of your changes:

**For Dune Engine V2**

I've checked that:

### General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by andrewhong5297** [comment]
@soispoke so I can get it running locally using `dbt run-operation get_chain_explorer --args '{"chain":'ethereum'}'`

Screen Shot 2022-11-10 at 10 15 41 AM

**Comment by soispoke** [comment]
@andrewhong5297 ok yeah definitely got it working as well, then the issue is just to having it accessible in the Dune App

**Comment by andrewhong5297** [comment]
> @andrewhong5297 ok yeah definitely got it working as well, then the issue is just to having it accessible in the Dune App

got it - any ideas?

**Comment by soispoke** [comment]
@andrewhong5297 we might want to try and use dbt prehooks to load macros when jobs are starting https://docs.getdbt.com/reference/resource-configs/pre-hook-post-hook

**Comment by andrewhong5297** [comment]
hmm anything I can do to help with this?

**Comment by soispoke** [comment]
yeah definitely, you can use this PR to set up a dbt prehook on your macros

**Comment by andrewhong5297** [comment]
okay, so based on my understanding we need prehooks that specifically create a UDF within the macro. 

then we can just add the pre-hook in `on-run-start:` in dbt_project.yml like they do here (https://www.equalexperts.com/blog/our-thinking/testing-and-deploying-udfs-with-dbt/)

only issue is I'm not sure where `create or replace function` permissions sit, and how it would work with the slim ci. @dot2dotseurat might need to pick your brain on this next week and test/add the UDFs manually through databricks for now.

**Comment by andrewhong5297** [comment]
okay yeah, I don't have permission to create or replace functions. Maybe we could create a schema that users do have permission to edit, like we did on v1 engine?

image

**Comment by andrewhong5297** [comment]
https://dune.com/queries/1627465 this one is now working, cc @dot2dotseurat

**Comment by dot2dotseurat** [comment]
Did you mean to close this?

**Comment by andrewhong5297** [comment]
whoops, looks like when I was merging conflicts I closed the PR somehow

File: dbt_project.yml [hunk 1]
```diff
@@ -10,6 +10,10 @@ profile: "spellbook"
 vars:
   DBT_ENV_CUSTOM_ENV_S3_BUCKET: "{{ env_var('DBT_ENV_CUSTOM_ENV_S3_BUCKET', 'local') }}"
 
+# # pre-run hooks for macros that create UDFs. commenting out for now.
```
Comment (dot2dotseurat) [inline dbt_project.yml:13]: does this comment need to be here?

File: macros/public/get_chain_native_token.sql [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{% macro get_chain_native_token() %}
+      create or replace function get_chain_native_token(chain_ STRING, column_ STRING)
```
Comment (dot2dotseurat) [inline macros/public/get_chain_native_token.sql:2]: very cool, elegant solution

File: macros/public/get_href.sql [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{% macro get_href() %}
+      create or replace function get_href(link STRING, text STRING)
```
Comment (dot2dotseurat) [inline macros/public/get_href.sql:2]: luv it

File: dbt_project.yml [hunk 4]
```diff
@@ -10,6 +10,9 @@ profile: "spellbook"
 vars:
   DBT_ENV_CUSTOM_ENV_S3_BUCKET: "{{ env_var('DBT_ENV_CUSTOM_ENV_S3_BUCKET', 'local') }}"
 
+# on-run-start:
```
Comment (dot2dotseurat) [inline dbt_project.yml:13]: it looks like it's still here

**Review by jeff-dude — APPROVED** [review]
lgtm after resolving her comments 🤝

**Review by dot2dotseurat — APPROVED** [review]

**Comment by bernatfp** [comment]
So cool @andrewhong5297

**Pull request merged**